test/balance_reader.spec.ts

```diff
@@ -0,0 +1,171 @@
+        erc20.address,
+        startingAmount.addn(1)
+      );
+      await closeAuction(batchExchange);
+      const withdrawableAmount = await getWithdrawableAmount(
+        accounts[0],
+        erc20.address,
+        batchExchange,
+        web3
+      );
+      assert.equal(withdrawableAmount.toString(), startingAmount.toString());
+      const transcript = await batchExchange.withdraw(
+        accounts[0],
+        erc20.address
+      );
+      assert.equal(
+        withdrawableAmount.toString(),
+        transcript.logs[0].args.amount.toString()
+      );
+    });
+    it("ignores deposits if funds are not yet available", async () => {
+      const erc20 = await MockContract.new();
+      await erc20.givenAnyReturnBool(true);
+      const batchExchange = await BatchExchange.new(1, erc20.address);
+      await closeAuction(batchExchange);
+      await batchExchange.deposit(erc20.address, startingAmount);
+      await applyBalances(accounts[0], batchExchange, [erc20.address]);
+
+      const newAmount = new BN("200");
+      const fullBalance = newAmount.add(startingAmount);
+      await batchExchange.requestWithdraw(erc20.address, fullBalance);
+      await closeAuction(batchExchange);
+      await batchExchange.deposit(erc20.address, newAmount);
+      const withdrawableAmount = await getWithdrawableAmount(
+        accounts[0],
+        erc20.address,
+        batchExchange,
+        web3
+      );
+      assert.equal(withdrawableAmount.toString(), startingAmount.toString());
+      const transcript = await batchExchange.withdraw(
+        accounts[0],
+        erc20.address
+      );
+      assert.equal(
+        withdrawableAmount.toString(),
+        transcript.logs[0].args.amount.toString()
+      );
+    });
+    it("considers deposits if funds are available", async () => {
+      const erc20 = await MockContract.new();
+      await erc20.givenAnyReturnBool(true);
+      const batchExchange = await BatchExchange.new(1, erc20.address);
+      await closeAuction(batchExchange);
+      await batchExchange.deposit(erc20.address, startingAmount);
+      await applyBalances(accounts[0], batchExchange, [erc20.address]);
+
+      const newAmount = new BN("200");
+      const fullBalance = newAmount.add(startingAmount);
+      await batchExchange.requestWithdraw(erc20.address, fullBalance);
+      await batchExchange.deposit(erc20.address, newAmount);
+      await closeAuction(batchExchange);
+      const withdrawableAmount = await getWithdrawableAmount(
+        accounts[0],
+        erc20.address,
+        batchExchange,
+        web3
+      );
+      assert.equal(withdrawableAmount.toString(), fullBalance.toString());
+      const transcript = await batchExchange.withdraw(
+        accounts[0],
+        erc20.address
+      );
+      assert.equal(
+        withdrawableAmount.toString(),
+        transcript.logs[0].args.amount.toString()
+      );
+    });
+    // untested case: lastCreditBatchId != 0
```

Comment:
Are these going to be added?

Reply:
No, it's more of a warning in case we find some issues in the future and look into the tests to see what's going on.